### PR TITLE
PT-82: Configuration of password expiration policy

### PR DESCRIPTION
**Pull request by eMazeika**
https://virtocommerce.atlassian.net/browse/PT-81

### CODE REVIEW

**Comment by mvktsk** [comment]
Review task https://virtocommerce.atlassian.net/browse/VP-7333 has been created

**Review by akak1977 — APPROVED** [review]

**Review by Vectorfield4 — APPROVED** [review]

File: src/VirtoCommerce.Platform.Core/Security/UserOptionsExtended.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using System;
+
+namespace VirtoCommerce.Platform.Core.Security
+{
+    public class UserOptionsExtended
+    {
+        /// <summary>
+        /// Max. valid time for a password. Property that has the null as default value. Null value means what the password never expires.
+        /// </summary>
+        public TimeSpan MaxPasswordAge { get; set; }
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Core/Security/UserOptionsExtended.cs:10]: Please make TimeSpan nullable if you decided to ignore my recommendation to name this setting as MaxPasswordAgeInDays

**Review by ksavosteev — APPROVED** [review]

File: src/VirtoCommerce.Platform.Web/Controllers/Api/AuthorizationController.cs [hunk 2]
```diff
@@ -84,6 +87,15 @@ public async Task<ActionResult> Exchange(OpenIdConnectRequest openIdConnectReque
                     });
                 }
 
+
+                // check and expire user password if needed
+                var lastPasswordChangeDate = user.LastPasswordChangedDate ?? user.CreatedDate;
```
Comment (tatarincev) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/AuthorizationController.cs:92]: Please move the code for evaluation the PasswordExpired to this place, and do not call update for user.
https://github.com/VirtoCommerce/vc-platform/blob/master/src/VirtoCommerce.Platform.Web/Security/CustomUserManager.cs#L287

**Review by tatarincev — CHANGES_REQUESTED** [review]
Also we need a write unit-tests for this logic

**Review by Vectorfield4 — APPROVED** [review]

**Review by tatarincev — APPROVED** [review]

**Pull request merged**